src/gti.jl

```diff
@@ -0,0 +1,98 @@
+function time_intervals_from_gtis(gtis::AbstractMatrix{<:Real}, segment_size::Real;
+                                  fraction_step::Real=1, epsilon::Real=1e-5)  
+    spectrum_start_times = Float64[]
+
+    gti_low = @view gtis[:,1]
+    gti_up = @view gtis[:,2]
+
+    for (g1,g2) in zip(gti_low,gti_up)
+        if g2 - g1 + epsilon < segment_size
+            continue
+        end
+
+        newtimes = range(g1, g2 - segment_size + epsilon, step=Float64(segment_size) * fraction_step)
+        append!(spectrum_start_times,newtimes)
+    end
+    return spectrum_start_times, spectrum_start_times .+ segment_size
+end
+
+function calculate_segment_bin_start(startbin::Integer, stopbin::Integer,
+                                     nbin::Integer; fraction_step::Real=1)
+    st = floor.(range(startbin, stopbin, step=Int64(nbin * fraction_step)))
+    if st[end] == stopbin
+        st = st[1:end-1]
+    end
+    if st[end] + nbin > stopbin
+        return st[1:end-1]
+    end
+    return st
+end
+
+function bin_intervals_from_gtis(gtis::AbstractMatrix{<:Real}, segment_size::Real,
+                                 time::AbstractVector{<:Real}; dt=nothing, 
+                                 fraction_step::Real=1, epsilon::Real=0.001)
+    if isnothing(dt)
+        dt = Statistics.median(diff(time))
+    end
+
+    epsilon_times_dt = epsilon * dt
+    nbin = Int64(round(segment_size / dt))
```

Comment:
```suggestion
    nbin = round(Int, segment_size / dt)
```

Reply:
But actually, integer division (already suggested in https://github.com/matteobachetti/Stingray.jl/pull/2#discussion_r905451223) is much idiomatic
```suggestion
    nbin = segment_size ÷ dt
```